Source/Core/VideoCommon/OnScreenDisplay.cpp

```diff
@@ -94,7 +95,7 @@ void AddMessage(std::string message, u32 ms, u32 rgba)
 
 void DrawMessages()
 {
-  const bool draw_messages = SConfig::GetInstance().bOnScreenDisplayMessages;
+  const bool draw_messages = Config::Get(Config::MAIN_OSD_MESSAGES);
```

Comment:
Is this a potentially time-sensitive location that could use some caching (and a hook into the config changed callback)? Feels like this might be called fairly often (like every frame or so).

Reply:
I believe this is called exactly once a frame. I've assumed that running this 60 times a second wouldn't have any noticeable impact.

Reply:
Thinking of #8884 somewhere in the back of my mind, which seemed similar. Didn't test (nor profile) it though, so I'm just guessing from staring at the code.